### PR TITLE
[Issue 549] Update JSON-RPC over HTTP error responses to return 200 status code 

### LINKED ISSUE #549: Update JSON-RPC over HTTP error responses to return 200 status code
JSON-RPC is supposed to be agnostic of what channel is being used. As per the 2.0 spec, there is no reason for using anything but 200 http status code whenever we are doing JSON-RPC over HTTP.

Also, returning 400 for valid JSON-RPC requests doesn't help when integrating with JSON-RPC clients.

Acceptance criteria

Any valid JSON-RPC request should generate a JSON-RPC response, the HTTP status code should be 200. No matter if the response is a success or error response.
Calling a method that hasn't been enabled or a method that doesn't exist are valid JSON-RPC requests, therefore the expected response is a JSON-RPC error (and status code = 200)
Any invalid JSON-RPC request (malformed json, etc) should generate a 400 status code.

### PR DESCRIPTION
**Pull request by shemnon**
# PR description

* Updated returned HTTP status code to 200 for JSON-RPC error
* Updated tests related to the modified HTTP status code

# Fixed Issue(s)
Was #1305, Fixes #549

# Changelog
- [X] I thought about the changelog and included a changelog update if required.

### CODE REVIEW

**Comment by shemnon** [comment]
This is a re-spin of #1305, with the disabling of some acceptance tests added in. Re-spun to ensure we can fast-track it into the RC1 build on Thursday.

 @gabrieledm did all the interesting work.

**Comment by shemnon** [comment]
Also, see issue #1425 reminding us to fix the Acceptance Tests.

**Review by macfarla — APPROVED** [review]
LGTM - just needs a changelog entry :)

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,17 @@
...
 
+## 20.10 Breaking Changes
+
+When upgrading to 20.10, ensure you've taken into account the following breaking changes.
+
+## JSON-RPC HTTP Error Codes For Valid Calls
+
+Prior versions of Besu would set the HTTP Status 400 Bad Request for JSON-RPC requests that completed in an error, regardless of the kind of error.  These responses could include a complete JSON-RPC response with an error field.
+
+In Besu version 20.10 properly formatted requests that have valid parameters (count and content) that otherwise would result in an error will return a HTTP Status 200 OK a JSON-RPC response with an error field will be returned.  For example, requesting an account that does not exist in the chain or a block by hash that Besu does not have will now return HTTP 200 OK responses.  Unparsable requests, improperly formatted requests, or requests with invalid parameters will continue to return HTTP 400 Bad Request.
```
Comment (macfarla) [inline CHANGELOG.md:11]: Suggestion: In Besu version 20.10, properly formatted requests that have valid parameters (count and content) will return a HTTP Status 200 OK, with an error field if an error occurred.  For example, requesting an account that does not exist in the chain, or a block by hash that Besu does not have, will now return HTTP 200 OK responses.  Unparsable requests, improperly formatted requests, or requests with invalid parameters will continue to return HTTP 400 Bad Request.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,17 @@
...
+
+This was done to bring us more in line with the behavior of other Ethereum Clients.  Some community project, such as Web3J versions prior to 4.6.x, will be providing compatible releases in the near future.
```
Comment (macfarla) [inline CHANGELOG.md:13]: Suggestion: This was done to bring us more in line with the behavior of other Ethereum Clients.  Some community projects, such as Web3J, will be providing compatible releases in the near future.

**Review by macfarla — COMMENTED** [review]
couple of rewording suggestions

**Pull request merged**